llvm/include/llvm/IR/Instructions.h

```diff
@@ -252,7 +252,8 @@ class LoadInst : public UnaryInstruction {
 
   /// Returns false if this type would be invalid in the
   /// creation of a load atomic instruction.
-  static bool isValidAtomicTy(Type *Ty);
+  static bool isValidAtomicTy(Type *Ty, const DataLayout *DL = nullptr,
```

Comment:
DataLayout must be mandatory

Reply:
Then we need to omit it entirely because otherwise we will be checking the type size of the alloca, which we want to avoid if to optimize `basictest.ll`.